modules/math/shared/src/main/scala/gsp/math/skycalc/ImprovedSkyCalc.scala

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gsp.math.skycalc
+
+import cats.Eq
+import cats.implicits._
+import gsp.math.Angle
+import gsp.math.Coordinates
+import java.time.Instant
+import java.time.ZonedDateTime
+
+/**
+  * Improved version of SkyCalc that supports lunar calculations. All instance stuff is here;
+  * the trait is exclusively static stuff.
+  * @author brighton, rnorris
+  */
+final class ImprovedSkyCalc(siteLatitude: Angle, siteLongitude: Angle, siteAlt: Int)
+    extends ImprovedSkyCalcMethods {
+
+  // Site parameters
+  private val hoursLongitude  = -siteLongitude.toDoubleDegrees / 15.0
+  private val degreesLatitude = siteLatitude.toDoubleDegrees
+  private val siteAltitude    = siteAlt.toDouble
+
+  // calculated results
+  private var altitude                             = .0
+  private var hourAngle                            = .0
+  private var azimuth                              = .0
+  private var parallacticAngle                     = .0
+  private var airmass                              = .0
+  private var lunarSkyBrightness: java.lang.Double = .0
+  private var lunarDistance                        = .0
+  private var lunarIlluminatedFraction             = .0
+  private var totalSkyBrightness                   = .0
+  private var lunarPhaseAngle                      = .0
+  private var sunAltitude                          = .0
+  private var lunarElevation                       = .0
+
+  // caching for calculate()
+  private var cachedCoordinates: Option[Coordinates] = none
+  private var cachedInstant: Option[Instant]         = none
+  private var cachedCalculateMoon: Option[Boolean]   = none
+
+  private implicit val eqInstant: Eq[Instant] = Eq.by(i => (i.getEpochSecond, i.getNano))
```

Comment:
I wonder if this is defined in some of the other libs. On the other hand it is private, we probably don't need to worry

Reply:
I considered using https://github.com/ChristopherDavenport/cats-time. I don't know if it's worth adding a dependency just for this.

Reply:
I think this is fine as-is, but you might check to see what `.equals` does for `Instant` … we may be able to define this as `Eq.fromUniversalEquals`.

Reply:
Ok. `Instant.equals` does the same thing. Changing it.